fearless/Common/Operation/EraCountdownOperationFactory/EraCountdown.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+
+struct EraCountdown {
+    let eraLength: SessionIndex
+    let sessionLength: SessionIndex
+    let eraStartSessionIndex: SessionIndex
+    let currentSessionIndex: SessionIndex
+    let currentSlot: Slot
+    let genesisSlot: Slot
+    let blockCreationTime: Moment
+    let createdAtDate: Date
+
+    var eraCompletionTime: TimeInterval {
+        let numberOfSlotsPerSession = UInt64(sessionLength)
+        let currentSessionIndexInt = UInt64(currentSessionIndex)
+
+        let sessionStartSlot = currentSessionIndexInt * numberOfSlotsPerSession + genesisSlot
+        let sessionProgress = currentSlot - sessionStartSlot
+        let eraProgress = (currentSessionIndexInt - UInt64(eraStartSessionIndex)) * numberOfSlotsPerSession
+            + sessionProgress
+        let eraRemained = UInt64(eraLength) * numberOfSlotsPerSession - eraProgress
```

Comment:
We might run into the situation when currentIndex already past current era but due to network lag we still on the old era. Let's put a check here before subtracting

Reply:
Now we manually fetch activeEra, so the these situation is impossible, right?

Reply:
I think yes. Because we don't fetch active era and currentSessionIndex atomically. So, it is better to add comparison and in case of negative result return 0